torch/csrc/utils/python_arg_parser.h

```diff
@@ -389,9 +390,46 @@ inline std::vector<int64_t> PythonArgs::intlist(int i) {
   return intlistWithDefault(i, signature.params[i].default_intlist);
 }
 
+static bool is_symint_node(py::handle obj) {
+      auto static tp_symn = py::type::of<c10::SymbolicIntNode>();
+      return obj.get_type().equal(tp_symn);
+}
+
 inline std::vector<c10::SymInt> PythonArgs::symintlist(int i) {
-  auto intlist = intlistWithDefault(i, signature.params[i].default_intlist);
-  return c10::fmap(intlist, [](int64_t n) {return c10::SymInt(n); });
+
+  if (!args[i]) {
+    return c10::fmap(signature.params[i].default_intlist, [](int64_t di) {
```

Comment:
It shouldn't be necessary to fmap here, `default_intlist` should be representationally equivalent to the sym intlist you want to return here. (I also wondered if a symintlist should store a `default_symintlist` but I suppose you don't actually want a symbolic integer in your defaults.)